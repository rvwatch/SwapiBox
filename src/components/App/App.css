.main-nav {
  position: absolute;
  width:100%;
  text-align: center;
  z-index: 5;
  top:20px;
  border-bottom: 3px solid #feda4a;
  padding-bottom: 20px;
}

.main-nav li {
  display: inline-block;
  margin: 0 20px;
}

.main-nav a {
  color: #fff;
  font-size: 30px;
  text-decoration: none;
}

.main-nav a:hover{
  color: #feda4a;
}

.main-nav li.title a{
  font-size: 60px;
  font-weight: lighter;
  color: #feda4a;
  position: absolute;
  left: 30px;
  top: -16px;
  -webkit-text-fill-color: black;
  -webkit-text-stroke-width: 2px;
  -webkit-text-stroke-color: feda4a;
}

@keyframes move-twink-back {
  from {background-position:0 0;}
  to {background-position:-10000px 5000px;}
}
@-webkit-keyframes move-twink-back {
  from {background-position:0 0;}
  to {background-position:-10000px 5000px;}
}
@-moz-keyframes move-twink-back {
  from {background-position:0 0;}
  to {background-position:-10000px 5000px;}
}
@-ms-keyframes move-twink-back {
  from {background-position:0 0;}
  to {background-position:-10000px 5000px;}
}

.stars, .twinkling, .clouds {
position:absolute;
top:0;
left:0;
right:0;
bottom:0;
display:block;
}

.stars {
background:#000 url(http://www.script-tutorials.com/demos/360/images/stars.png) repeat top center;
z-index:0;
}

.twinkling{
background:transparent url(http://www.script-tutorials.com/demos/360/images/twinkling.png) repeat top center;
z-index:1;

-moz-animation:move-twink-back 200s linear infinite;
-ms-animation:move-twink-back 200s linear infinite;
-o-animation:move-twink-back 200s linear infinite;
-webkit-animation:move-twink-back 200s linear infinite;
animation:move-twink-back 200s linear infinite;
}

