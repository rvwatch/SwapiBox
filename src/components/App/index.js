import React, { Component } from 'react';
import Home from '../Home';
import Main from '../Main';
import { Link } from 'react-router-dom';
import './App.css';

class App extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <div>
       
        <div className="stars"></div>
        <div className="twinkling"></div>
        <ul className='main-nav'>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/People">People</Link>
          </li>
          <li>
            <Link to="/Planets">Planets</Link>
          </li>
          <li>
            <Link to="/Vehicles">Vehicles</Link>
          </li>
          <li>
            <Link to="/Favorites">Favorites</Link>
          </li>
        </ul>
        <Main />
      </div>
    );
  }
}

export default App;
