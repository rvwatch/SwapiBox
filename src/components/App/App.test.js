import React from 'react';
import { shallow } from 'enzyme';
import App from './index';

describe('App ', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

});