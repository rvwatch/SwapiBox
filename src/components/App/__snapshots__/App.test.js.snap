// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App  renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="stars"
/>,
        <div
          className="twinkling"
/>,
        <ul
          className="main-nav"
>
          <li
                    className="title"
          >
                    <Link
                              replace={false}
                              to="/"
                    >
                              SWAPI BOX
                    </Link>
          </li>
          <li>
                    <Link
                              replace={false}
                              to="/People"
                    >
                              People
                    </Link>
          </li>
          <li>
                    <Link
                              replace={false}
                              to="/Planets"
                    >
                              Planets
                    </Link>
          </li>
          <li>
                    <Link
                              replace={false}
                              to="/Vehicles"
                    >
                              Vehicles
                    </Link>
          </li>
          <li>
                    <Link
                              replace={false}
                              to="/Favorites"
                    >
                              Favorites
                    </Link>
          </li>
</ul>,
        <Main />,
      ],
      "className": "app-wrap",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stars",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "twinkling",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="title"
>
              <Link
                            replace={false}
                            to="/"
              >
                            SWAPI BOX
              </Link>
</li>,
            <li>
              <Link
                            replace={false}
                            to="/People"
              >
                            People
              </Link>
</li>,
            <li>
              <Link
                            replace={false}
                            to="/Planets"
              >
                            Planets
              </Link>
</li>,
            <li>
              <Link
                            replace={false}
                            to="/Vehicles"
              >
                            Vehicles
              </Link>
</li>,
            <li>
              <Link
                            replace={false}
                            to="/Favorites"
              >
                            Favorites
              </Link>
</li>,
          ],
          "className": "main-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
>
                SWAPI BOX
</Link>,
              "className": "title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "SWAPI BOX",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "SWAPI BOX",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/People"
>
                People
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "People",
                "replace": false,
                "to": "/People",
              },
              "ref": null,
              "rendered": "People",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/Planets"
>
                Planets
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Planets",
                "replace": false,
                "to": "/Planets",
              },
              "ref": null,
              "rendered": "Planets",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/Vehicles"
>
                Vehicles
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Vehicles",
                "replace": false,
                "to": "/Vehicles",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/Favorites"
>
                Favorites
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Favorites",
                "replace": false,
                "to": "/Favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stars"
/>,
          <div
            className="twinkling"
/>,
          <ul
            className="main-nav"
>
            <li
                        className="title"
            >
                        <Link
                                    replace={false}
                                    to="/"
                        >
                                    SWAPI BOX
                        </Link>
            </li>
            <li>
                        <Link
                                    replace={false}
                                    to="/People"
                        >
                                    People
                        </Link>
            </li>
            <li>
                        <Link
                                    replace={false}
                                    to="/Planets"
                        >
                                    Planets
                        </Link>
            </li>
            <li>
                        <Link
                                    replace={false}
                                    to="/Vehicles"
                        >
                                    Vehicles
                        </Link>
            </li>
            <li>
                        <Link
                                    replace={false}
                                    to="/Favorites"
                        >
                                    Favorites
                        </Link>
            </li>
</ul>,
          <Main />,
        ],
        "className": "app-wrap",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stars",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "twinkling",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="title"
>
                <Link
                                replace={false}
                                to="/"
                >
                                SWAPI BOX
                </Link>
</li>,
              <li>
                <Link
                                replace={false}
                                to="/People"
                >
                                People
                </Link>
</li>,
              <li>
                <Link
                                replace={false}
                                to="/Planets"
                >
                                Planets
                </Link>
</li>,
              <li>
                <Link
                                replace={false}
                                to="/Vehicles"
                >
                                Vehicles
                </Link>
</li>,
              <li>
                <Link
                                replace={false}
                                to="/Favorites"
                >
                                Favorites
                </Link>
</li>,
            ],
            "className": "main-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/"
>
                  SWAPI BOX
</Link>,
                "className": "title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "SWAPI BOX",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "SWAPI BOX",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/People"
>
                  People
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "People",
                  "replace": false,
                  "to": "/People",
                },
                "ref": null,
                "rendered": "People",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/Planets"
>
                  Planets
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Planets",
                  "replace": false,
                  "to": "/Planets",
                },
                "ref": null,
                "rendered": "Planets",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/Vehicles"
>
                  Vehicles
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Vehicles",
                  "replace": false,
                  "to": "/Vehicles",
                },
                "ref": null,
                "rendered": "Vehicles",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/Favorites"
>
                  Favorites
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Favorites",
                  "replace": false,
                  "to": "/Favorites",
                },
                "ref": null,
                "rendered": "Favorites",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
